src/Microsoft.Azure.CosmosEventSourcing/Converters/PersistedEventConverter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.Cosmos;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Converters;
+
+public class PersistedEventConverter : JsonConverter
+{
+    public static HashSet<Type> ConvertableTypes { get; } = new();
+
+    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer) =>
+        serializer.Serialize(writer, value, value.GetType());
+
+    public override object ReadJson(
+        JsonReader reader,
+        Type objectType,
+        object existingValue,
+        JsonSerializer serializer)
+    {
+        JToken? j = JToken.ReadFrom(reader);
```

Comment:
@IEvangelist do you know a better way to do this? 

I'd be happy to use System.Text if it makes it easier.

Reply:
Okay so been trying a few things - this seems to be the most efficient

https://github.com/mumby0168/PolymorphicDotnetJson/blob/b98250ddede01fcf69cb257a64c6f6615a87986f/src/PolymorphicDotnetJson/SystemText/SystemTextEventConverter.cs#L6

That is baed on these benchmarks

The first uses what I have now with a `JObject` the other uses a `JsonDocument` and the final one uses a raw parse of the JSON which I thought would be better but appears not.

<img width="836" alt="Screenshot 2022-02-05 at 17 22 44" src="https://user-images.githubusercontent.com/23740684/152651980-8dc43d28-29fe-45fa-9700-88e0ea628c41.png">

Reply:
This is exciting! I'm not well versed in event sourcing architectures, but just a high-level comment that if we're looking for inspiration from other impl's in document dbs, there's Marten by Jeremy Miller: https://github.com/JasperFx/marten

Reply:
Ahhh yeah that's very cool! I'd heard of this before, but I haven't properly looked at it. Thanks @dcuccia